client/src/components/DisplayTable/DisplayTable.tsx

```diff
@@ -274,51 +277,53 @@ export class _DisplayTable extends React.Component<
       _.chain(search_string).deburr().toLower().trim().value();
     const is_number_string_date = (val: number | string | Date) =>
       _.isNumber(val) || _.isString(val) || _.isDate(val);
-    const sorted_filtered_data = _.chain(data)
-      .filter((row) =>
-        _.chain(row)
-          .map((column_value: CellValue, column_key) => {
-            const col_config = col_configs_with_defaults[column_key];
-            const col_search_value =
-              col_config && col_config.raw_formatter
-                ? col_config.raw_formatter(column_value)
-                : column_value;
-            return (
-              _.isEmpty(searches[column_key]) ||
-              _.includes(
-                clean_search_string(col_search_value),
-                clean_search_string(searches[column_key])
-              )
-            );
-          })
-          .every()
-          .value()
-      )
-      .thru((unsorted_array) => {
-        if (sort_by && _.has(col_configs_with_defaults, sort_by)) {
-          const sorting_config = col_configs_with_defaults[sort_by];
-          return sorting_config.sort_func
-            ? _.map(unsorted_array).sort(
-                (a: DisplayTableData, b: DisplayTableData) =>
-                  sorting_config.sort_func(a[sort_by], b[sort_by], descending)
-              )
-            : _.sortBy(unsorted_array, (row: DisplayTableData) =>
-                is_number_string_date(
-                  sorting_config.raw_formatter(row[sort_by])
-                ) /*Please Leave On: sorting data containing BOTH string and date is not consistent
+
+    const sorted_filtered_data = (data: DisplayTableData[]) =>
```

Comment:
- this should be renamed as a _function_ now like `sort_and_filter_data`
- But before doing that, I don't think this needs to be a function. Can you just reorder and group the data before getting `total_row`, `csv_string`, etc, so then group the data first, sort and filter the data, then do everything else? Haven't looked at it thorough enough to see if that's possible so lemme know

Reply:
Yep, it can be reordered